src/dbnode/storage/index/aggregate_results.go

```diff
@@ -52,15 +49,21 @@ func NewAggregateResults(
 	aggregateOpts AggregateResultsOptions,
 	opts Options,
 ) AggregateResults {
-	return &aggregatedResults{
+	results := &aggregatedResults{
 		nsID:          namespaceID,
 		aggregateOpts: aggregateOpts,
-		resultsMap:    newAggregateResultsMap(opts.IdentifierPool()),
-		idPool:        opts.IdentifierPool(),
-		bytesPool:     opts.CheckedBytesPool(),
-		pool:          opts.AggregateResultsPool(),
-		valuesPool:    opts.AggregateValuesPool(),
 	}
+
+	if opts != nil {
+		results.idPool = opts.IdentifierPool()
+		results.pool = opts.AggregateResultsPool()
+		results.valuesPool = opts.AggregateValuesPool()
+		results.resultsMap = newAggregateResultsMap(opts.IdentifierPool())
+	} else {
+		results.resultsMap = newAggregateResultsMap(nil)
+	}
```

Comment:
Same with this paranoia, would prefer `opts` is always specified or crashes during integration tests.